demos/common/img_cap.h

```diff
@@ -0,0 +1,145 @@
+#include <stdexcept>
+#include <string>
+#include <memory>
+
+#ifdef _WIN32
+#include <os/windows/w_dirent.h>
+#else
+#include <dirent.h>
+#endif
+
+#include <opencv2/core/mat.hpp>
+#include <opencv2/imgcodecs.hpp>
+#include <opencv2/videoio.hpp>
+
+class InvalidInput : public std::invalid_argument {
+public:
+    InvalidInput() : std::invalid_argument{""} {}
```

Comment:
`InvalidInput` is used only to report inability to open an input. I never use its message.

Reply:
Then why does it need to inherit from `std::invalid_argument`?

Reply:
It feels that failing to open an input with a specific reader has a semantic of `invalid_argument` for that reader.

Reply:
`invalid_argument` is a subtype of `logic_error`, which is generally used to indicate an error in the program. Your exception is much closer to `runtime_error`.

With that said, the main purpose in inheriting your exception class from _anything_ is to enable it to be caught as a base class reference, and since you never do that, you don't need any inheritance.

Reply:
Updated